contracts/src/v0.8/ccip/DonIdClaimer.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {ICapabilityConfiguration} from "../keystone/interfaces/ICapabilityConfiguration.sol";
+
+contract DonIDClaimer {
+  error ZeroAddressNotAllowed();
+
+  uint256 public donID;  
+  address internal immutable i_capabilitiesRegistry;
+
+  constructor(
+    address capabilitiesRegistry
+  ) {
+    if (capabilitiesRegistry == address(0)) {
+      revert ZeroAddressNotAllowed();
+    }
+    i_capabilitiesRegistry = capabilitiesRegistry;
+  }
+
+  function claimNextDonID() external returns (uint256) {
+    donID = donID + 1;
+    return donID; 
+  }
+
+  function setDONID(uint256 donId) external pure returns (uint256) {
+      return donId; 
+  }
+
+  function getDonID() external view returns (uint256) {
+      return donID; 
+  }
```

Comment:
I would remove these, especially the setter as it would provide a unwanted side entrance. We only need the claim function.

If we want a getter, we should align the API with that of the CapabilitiesRegistry: `function getNextDONId() external view returns (uint32)`



```suggestion
  function setDONID(uint256 donId) external pure returns (uint256) {
      return donId; 
  }

  function getDonID() external view returns (uint256) {
      return donID; 
  }
```

Reply:
`setDONID` was added as a worst-case fallback mechanism to manually reset donID if it gets super out of sync.

For `getDonID` i can align with the function sig from CapReg. But wanted to keep it as it makes retrieving this value simpler especially in tooling

Reply:
https://github.com/smartcontractkit/chainlink/pull/16926/commits/e5e4a9ed063b98f455d7792e02a2adf5da36ca33